### PR TITLE
preact create command template listing fixes.

### PR DESCRIPTION
**Pull request by reznord**
**What kind of change does this PR introduce?**
refactoring

**Did you add tests for your changes?**
Not required

**Summary**
Fixes the comments mentioned here: https://github.com/preactjs/preact-cli/pull/1035#pullrequestreview-371045906

**Does this PR introduce a breaking change?**
No

### CODE REVIEW

File: packages/cli/lib/constants.js [hunk 1]
```diff
@@ -40,4 +42,4 @@ exports.FALLBACK_TEMPLATE_OPTIONS = [
 	},
 ];
 
-exports.TEMPLATES_CACHE_FILENAME = '.preact-templates.json';
+exports.TEMPLATES_CACHE_FILENAME = `${os.homedir()}/.cache/preact-templates.json`;
```
Comment (prateekbh) [inline packages/cli/lib/constants.js]: use `path.join`. also, not sure if `.cache` exist in all OS.

Comment (ForsakenHarmony) [reply packages/cli/lib/constants.js]: Yeah have to create if it doesn't

File: packages/cli/lib/commands/create.js [hunk 2]
```diff
@@ -96,12 +96,17 @@ function requestParams(argv, templates) {
...
+	try {
+		const repos = await fetch(TEMPLATES_REPO_URL).then(r => r.json());
+		await fs.writeFile(
+			TEMPLATES_CACHE_FILENAME,
+			JSON.stringify(repos, null, 2),
+			'utf-8'
+		);
+	} catch (err) {
+		// eslint-disable-next-line
+		console.log(err);
```
Comment (ForsakenHarmony) [inline packages/cli/lib/commands/create.js]:
```suggestion
		console.log('failed to update template cache', err);
```

Comment (ForsakenHarmony) [reply packages/cli/lib/commands/create.js]: Should maybe be `console.error` as well?

Comment (prateekbh) [reply packages/cli/lib/commands/create.js]: should use utils?

File: packages/cli/lib/constants.js (hunk 1 above)
Comment (reznord) [reply packages/cli/lib/constants.js]: done

File: packages/cli/lib/commands/create.js [hunk 3]
```diff
@@ -96,26 +98,35 @@ function requestParams(argv, templates) {
 }
 
 async function updateTemplatesCache() {
-	const repos = await fetch(TEMPLATES_REPO_URL).then(r => r.json());
-	await fs.writeFile(
-		TEMPLATES_CACHE_FILENAME,
-		JSON.stringify(repos, null, 2),
-		'utf-8'
-	);
+	const cacheFilePath = `${os.homedir()}/${TEMPLATES_CACHE_FOLDER}/${TEMPLATES_CACHE_FILENAME}`;
```
Comment (prateekbh) [inline packages/cli/lib/commands/create.js]: `path.join` instead of `/`

File: packages/cli/lib/commands/create.js [hunk 4]
```diff
@@ -96,26 +98,35 @@ function requestParams(argv, templates) {
...
+		const repos = await fetch(TEMPLATES_REPO_URL).then(r => r.json());
+		await fs.writeFile(cacheFilePath, JSON.stringify(repos, null, 2), 'utf-8');
+	} catch (err) {
+		error(`\nFailed to update template cache\n ${err}`);
+	}
 }
 
 async function fetchTemplates() {
 	let templates = [];
+	const cacheFolder = `${os.homedir()}/${TEMPLATES_CACHE_FOLDER}`;
```
Comment (prateekbh) [inline packages/cli/lib/commands/create.js]: path.join

File: packages/cli/lib/commands/create.js [hunk 5]
```diff
@@ -96,26 +98,35 @@ function requestParams(argv, templates) {
...
+	const cacheFilePath = `${cacheFolder}/${TEMPLATES_CACHE_FILENAME}`;
```
Comment (prateekbh) [inline packages/cli/lib/commands/create.js]: path.join

File: packages/cli/lib/commands/create.js (hunk 3 above)
Comment (reznord) [reply packages/cli/lib/commands/create.js]: done

File: packages/cli/lib/commands/create.js (hunk 5 above)
Comment (reznord) [reply packages/cli/lib/commands/create.js]: done

**Pull request merged**